ui-admin/src/util/tableUtils.tsx

```diff
@@ -36,6 +37,16 @@ function DebouncedInput({
   )
 }
 
+declare module '@tanstack/table-core' {
+  //Extra column metadata for extending the built-in filter functionality of react-table
+  interface ColumnMeta<TData extends RowData, TValue> {
+    //Specifies the type of the column data. By default, columns will be treated as strings
+    columnType?: string
+    //Specifies the Select options if using a dropdown filter (i.e. for booleans)
+    filterOptions?: object[]
+  }
+}
+
```

Comment:
I'm not certain of where this should actually go. Also the linter warns that `TData` and `TValue` are unused, but they seem to be needed to get this to extend properly?

Reply:
Yeah, I think ESLint isn't realizing that this has a side effect of modifying the `ColumnMeta` interface.

That being said... is there a way to avoid modifying that interface? Does Table take a type parameter for column metadata?

Reply:
I've been looking, but modifying the interface was how the developer of react-table suggested doing it: https://tanstack.com/table/v8/docs/api/core/column-def#meta